modules/core/shared/src/main/scala/lucuma/core/model/ConstraintSet.scala

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.model
+
+import cats._
+import eu.timepit.refined.cats._
+import lucuma.core.enum.CloudExtinction
+import lucuma.core.enum.ImageQuality
+import lucuma.core.enum.SkyBackground
+import lucuma.core.enum.WaterVapor
+import monocle.Focus
+import monocle.Lens
+import monocle.Optional
+
+/** Constraints for an observation. */
+final case class ConstraintSet(
+  imageQuality:    ImageQuality,
+  cloudExtinction: CloudExtinction,
+  skyBackground:   SkyBackground,
+  waterVapor:      WaterVapor,
+  elevationRange:  ElevationRange
```

Comment:
Do we ever let not having an elevation range? Or do we have the equivalent to `Any` range

Reply:
Hmm good question. `Airmass.Default` (1-2) seems fine to me.